---
title: "Sleuth Analysis"
author: "Ruben Sancho, Bruno Contreras Moreira"
output:
  html_document: default
  pdf_document: default
---

```{r knitr setup, include=FALSE,  eval=TRUE, echo=FALSE, warning=FALSE}
library(knitr)
knitr::opts_chunk$set(eval=TRUE, cache=FALSE, message=FALSE, warning=FALSE, 
                      comment = "")
```


```{r packages}

workingDir = ".";
setwd(workingDir);

suppressMessages({
  library('cowplot')
  library('sleuth')
})

```

```{r input_sleuth}

# Each file of single-end (SE) reads was mapped on Bdistachyon_314_v3.1.transcript.TagSeq500b.fa using --single kalllisto, -l 100 -s 20 

# The first step in a sleuth analysis is to specify where the kallisto results are stored. A variable is created for this purpose with

sample_id <- dir(file.path("kallisto_path", "kallisto_folder"))
sample_id

# e.g. [1] "BA002_HD_BdTR9k_GACACA"  "BA003_HD_BdTR2b_GAGTGG"  "BA005_HW_Bd30-1_GAAGTT"  "BA006_HD_ABR3_GCAGGA"    "BA007_HW_Bd3-1_GTATTA"


# A list of paths to the kallisto results indexed by the sample IDs is collated with

kal_dirs <- file.path("kallisto_path", "kallisto_folder", sample_id)
kal_dirs

# e.g. [1] "/kallisto_path/kallisto_folder/BA002_HD_BdTR9k_GACACA" 

# The next step is to load an auxillary table that describes the experimental design and the relationship between the kallisto directories and the samples

# sample_to_covariates (s2c)

s2c <- read.table('metadata.txt', header = TRUE, sep = '\t')
head(s2c)

s2c <- dplyr::select(s2c, sample = sample_id, genotype = genotype, date = date, condition = condition, temperature = temperature)
head(s2c)

# Now the directories must be appended in a new column to the table describing the experiment. This column must be labeled path, otherwise sleuth will report an error. This is to ensure that samples can be associated with kallisto quantifications.

s2c <- dplyr::mutate(s2c, path = kal_dirs)

# e.g.
# sample                 genotype  date    condition  temperature path
# BA002_HD_BdTR9k_GACACA   BdTR9k    a       Dry         hot      /kallisto_path/kallisto_folder/BA002_HD_BdTR9k_GACACA
```

```{r sleuth_prep_condition+date}

# Running in shell to multithreating

options(mc.cores = 6L)

so <- sleuth_prep(s2c,~condition+date, extra_bootstrap_summary = TRUE, num_cores = 6)

```

```{r run_sleuth_fit}

so <- sleuth_fit(so)

# Reduced model including the batch still

so <- sleuth_fit(so, ~date, 'reduced')

```

```{r run_sleuth_lrt}

so <- sleuth_lrt(so, 'reduced', 'full')

models(so)

save(so, file = "so_final.RData")

```

```{r view_so}

load(file = "so_final.RData")

sleuth_live(so)

```


```{r kallisto_sleuth_table_normalize}

kallisto_table_so_prep_condition_date <- kallisto_table(so, use_filtered =TRUE, normalized = TRUE, include_covariates = TRUE)

write.table(kallisto_table_so_prep_condition_date, file="sleuth_result.tsv", sep = "\t")

```

```{r sleuth_resuls}

sr <- sleuth_results(so, 'reduced:full', 'lrt')

write.table(sr, file="sr_final.txt")

```

```{r sr_0.000001}

sr_filter <- table(sr$qval <= 0.000001)

sr_filter
```
